### DISCUSSION TITLE
Run time with `--longitudinal`

### DISCUSSION THREAD

**Discussion by cookpa** [nipreps/fmriprep#2774] [Q&A] [answered]
Hi,

I'm trying to figure out how run time might scale for three longitudinal sessions, given a baseline for a single session. In the docs, I read

>In the case of multiple T1w images (across sessions and/or runs), T1w images are merged into a single template image using FreeSurfer’s [mri_robust_template](https://surfer.nmr.mgh.harvard.edu/fswiki/mri_robust_template). This template may be unbiased, or equidistant from all source images, or aligned to the first image (determined lexicographically by session label). For two images, the additional cost of estimating an unbiased template is trivial and is the default behavior, but, for greater than two images, the cost can be a slowdown of an order of magnitude. Therefore, in the case of three or more images, fMRIPrep constructs templates aligned to the first image, unless passed the --longitudinal flag, which forces the estimation of an unbiased template.

I've run `mri_robust_template` before, and it typically takes well under an hour, which is less than 10% of my usual run time for a single session. Does `--longitudinal` imply other processing in fmriprep will also take longer? I'm not sure where an order of magnitude slowdown comes from.

**Comment by effigies** [comment, marked as answer]
It's an iterative process of trying to find a rigid transform that is equidistant between the original three images. I would have to dig back into the `mri_robust_template` docs/paper to recall the exact process. The order-of-magnitude estimate was just based on observation of runtimes, I think.

**Reply by cookpa** [reply]
Oh, so is the above paragraph saying that `mri_robust_template` is an order of magnitude slower than aligning all session T1w to the first one? That fits with my observations. So then it's not saying overall run time would be an order of magnitude slower, just that specific step.

**Reply by effigies** [reply]
Yes that's right. Apologies for the ambiguity.

**Reply by cookpa** [reply]
No problem, thanks for clarifying!
